File: typescript-sdk/packages/core/src/data/repositories/image/image-repository.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

import type { Result } from '../../../common/result.js';
import type { ImageData } from '../../../domain/image-data.js';

export interface ImageRepository {
  /** Opens file picker and returns selected image data, or undefined if cancelled. */
  pickImage(): Promise<Result<ImageData | undefined>>;

  /** Saves an image and returns its URL / path. */
  save(fileName: string, data: Uint8Array, mimeType: string): Promise<Result<string>>;

  /** Deletes a saved image. */
  delete(fileName: string): Promise<Result<void>>;
}

File: android-sdk/sdk/src/commonMain/kotlin/com/yalo/chat/sdk/domain/repository/ImageRepository.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.domain.repository

import com.yalo.chat.sdk.common.Result
import com.yalo.chat.sdk.domain.model.ImageData

// Cancelled picker returns Result.Error — never null.
interface ImageRepository {
    suspend fun pickFromGallery(): Result<ImageData>
    suspend fun pickFromCamera(): Result<ImageData>
}
